From 741432351902b5548b5c441cb8dcec635782dfed Mon Sep 17 00:00:00 2001
From: Ryan Lortie <desrt@desrt.ca>
Date: Thu, 26 Sep 2013 17:37:46 +0000
Subject: GtkTreeViewColumn: start padding at 0

If we start with a padding of -1 then it can leak out of the size
allocation request for the column when the treeview is empty.  The
GtkTreeView will then collect these -1 values and add them together,
returning -n where 'n' is the number of columns.

This is usually not a problem because treeviews tend to be used with a
scrollbar and the width of the scrollbar will be added to this number
bringing it into positive territory again.  On Ubuntu, with overlay
scrollbars, this is not the case, however.

https://bugzilla.gnome.org/show_bug.cgi?id=703062
---
diff --git a/gtk/gtktreeviewcolumn.c b/gtk/gtktreeviewcolumn.c
index ea16706..e1846d5 100644
--- a/gtk/gtktreeviewcolumn.c
+++ b/gtk/gtktreeviewcolumn.c
@@ -464,7 +464,7 @@ gtk_tree_view_column_init (GtkTreeViewColumn *tree_column)
   priv->button = NULL;
   priv->xalign = 0.0;
   priv->width = 0;
-  priv->padding = -1;
+  priv->padding = 0;
   priv->min_width = -1;
   priv->max_width = -1;
   priv->column_type = GTK_TREE_VIEW_COLUMN_GROW_ONLY;

